contrib/src/test/java/gov/nasa/jpl/aerie/contrib/serialization/mappers/RecordValueMapperTest.java

```diff
@@ -0,0 +1,182 @@
+                )))))
+    ))).getSuccessOrThrow();
+
+    assertEquals(
+        new MultiValueRecord("one small step for crayon", Map.of("x", List.of(false), "y", List.of(true))),
+        deserializedValue);
+  }
+
+  @Test
+  void serializeValue_emptyRecord() {
+    final var serializedValue = new RecordValueMapper<>(EmptyRecord.class, List.of())
+        .serializeValue(new EmptyRecord());
+    assertEquals(SerializedValue.of(Map.of()), serializedValue);
+  }
+
+  @Test
+  void serializeValue_stringRecord() {
+    final var serializedValue = new RecordValueMapper<>(
+        StringRecord.class,
+        List.of(new RecordValueMapper.Component<>(
+            "aString",
+            StringRecord::aString,
+            new StringValueMapper()
+        ))).serializeValue(new StringRecord("hello world"));
+    assertEquals(SerializedValue.of(Map.of("aString", SerializedValue.of("hello world"))), serializedValue);
+  }
+
+  @Test
+  void serializeValue_multiValueRecord() {
+    final var serializedValue = new RecordValueMapper<>(
+        MultiValueRecord.class,
+        List.of(
+            new RecordValueMapper.Component<>(
+                "aString",
+                MultiValueRecord::aString,
+                new StringValueMapper()
+            ),
+            new RecordValueMapper.Component<>(
+                "fancy",
+                MultiValueRecord::fancy,
+                new MapValueMapper<>(new StringValueMapper(), new ListValueMapper<>(new BooleanValueMapper()))
+            ))).serializeValue(new MultiValueRecord(
+        "one small step for crayon",
+        Map.of("x", List.of(false), "y", List.of(true))));
+    final var permutation1 = SerializedValue.of(List.of(
+        SerializedValue.of(Map.of(
+            "key", SerializedValue.of("x"),
+            "value", SerializedValue.of(List.of(SerializedValue.of(false))
+            ))),
+        SerializedValue.of(Map.of(
+            "key", SerializedValue.of("y"),
+            "value", SerializedValue.of(List.of(SerializedValue.of(true))
+            )))));
+    final var permutation2 = SerializedValue.of(List.of(
+        SerializedValue.of(Map.of(
+            "key", SerializedValue.of("y"),
+            "value", SerializedValue.of(List.of(SerializedValue.of(true))
+            ))),
+        SerializedValue.of(Map.of(
+            "key", SerializedValue.of("x"),
+            "value", SerializedValue.of(List.of(SerializedValue.of(false))
+            )))));
+    if (!(SerializedValue.of(Map.of(
+        "aString", SerializedValue.of("one small step for crayon"),
+        "fancy", permutation1)).equals(serializedValue) || SerializedValue.of(Map.of(
+        "aString", SerializedValue.of("one small step for crayon"),
+        "fancy", permutation2)).equals(serializedValue))) {
+      assertEquals(SerializedValue.of(Map.of(
+          "aString", SerializedValue.of("one small step for crayon"),
+          "fancy", permutation1)), serializedValue);
+    } else {
+      //pass
```

Comment:
Why this empty block ?

Reply:
The problem I ran into was that the Map would be serialized in a non-deterministic order, so this test checks equality with both possible orders. I've updated the comment to explain this.